velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp

```diff
@@ -298,6 +309,7 @@ class ParquetTableScanTest : public HiveConnectorTestBase {
 
   RowTypePtr rowType_;
   std::vector<std::shared_ptr<connector::ConnectorSplit>> splits_;
+  TimestampPrecision timestampPrecision_ = TimestampPrecision::kMicroseconds;
```

Comment:
Why not use TimestampUnit here? We can avoid the conversion above.

Reply:
I would prefer we change parquet::WriterOptions to use Velox's TimestampPrecision instead of the arrow Bridge TimestampUnit.

Reply:
`parquet Writer has to do the conversion from TimestampPrecision in parquet::WriterOptions to TimestampUnit in arrow options.`

This works. The Parquet reader and writer APIs must be consistent.

Reply:
`timestamp-unit` in writer is already a session and a Hive connector variable.

https://github.com/facebookincubator/velox/blob/b44ffc9e9e85e3a0d21ec72f2e463ff17be011ee/velox/dwio/parquet/writer/Writer.h#L105-L119

If we change `TimestampUnit` in `parquet::WriterOptions` to `TimestampPrecision`, we have to depreciate this flag, and introduce a new flag `timestamp-precision`. Do you still want this sort of changes?

Reply:
> If we change TimestampUnit in parquet::WriterOptions to TimestampPrecision, we have to depreciate this flag, and introduce a new flag timestamp-precision

We can keep the config names as is. The read config also uses `hive.reader.timestamp-unit`.